guix-science/packages/geant4.scm

```diff
@@ -0,0 +1,699 @@
+nuclear and accelerator physics, as well as studies
+in medical and space science.
+
+This package supports visualisation with OpenGL and Qt.")))
+
+(define-public geant4-vis-11-1-2
+  (package
+    (inherit geant4-vis-11-1-1)
+    (version "11.1.2")
+    (source
+     (origin
+       (inherit (package-source geant4-vis-11-1-1))
+       (uri (git-reference
+             (url "https://gitlab.cern.ch/geant4/geant4")
+             (commit (string-append "v" version))))
+       (sha256
+        (base32 "03dbq48sycw8bfsf2vm6kk9f7ri663lfiw3blwigck0g4f1vdmhd"))))))
+
+(define-public geant4-vis-11-1-3
+  (package
+    (inherit geant4-vis-11-1-1)
+    (version "11.1.3")
+    (source
+     (origin
+       (inherit (package-source geant4-vis-11-1-1))
+       (uri (git-reference
+             (url "https://gitlab.cern.ch/geant4/geant4")
+             (commit (string-append "v" version))))
+       (sha256
+        (base32 "0w616j4w8gmmn5815pnyylm9sv0gpdicpjbvxwpyhzqlssmm2bjl"))))))
+
+;; -- Geant4 version 11.2.x --
+;; bump versions of CLHEP, G4EMLOW, G4ABLA, G4INCL
+
+(define-public geant4-11-2-0
+  (package
+    (inherit geant4-11-1-1)
+    (version "11.2.0")
+    (source
+     (origin
+       (inherit (package-source geant4-11-1-1))
+       (uri (git-reference
+             (url "https://gitlab.cern.ch/geant4/geant4")
+             (commit (string-append "v" version))))
+       (sha256
+        (base32 "1nblqmpp45k080rkrpb214q02c0d7qpwbqr9phr38vvaqndd710n"))))
+    (inputs (list coreutils
+                  gcc-toolchain
+                  xerces-c ;pour GDML
+                  expat
+                  clhep-2.4.7.1
+                  python-2
+                  python-3.10
+                  perl
+                  tcsh))
+    (native-inputs `(("G4NDL" ,g4ndl-4.7)
+                     ("G4EMLOW" ,g4emlow-8.5)
+                     ("G4PhotonEvaporation" ,photon-evaporation-5.7)
+                     ("G4RadioactiveDecay" ,radioactive-decay-5.6)
+                     ("G4PARTICLEXS" ,g4particlexs-4.0)
+                     ("G4PII" ,g4pii-1.3)
+                     ("G4RealSurface" ,real-surface-2.2)
+                     ("G4SAIDDATA" ,g4saiddata-2.0)
+                     ("G4ABLA" ,g4abla-3.3)
+                     ("G4INCL" ,g4incl-1.2)
+                     ("G4ENSDFSTATE" ,g4ensdfstate-2.3)
+                     ("G4TENDL" ,g4tendl-1.4)))))
+
+(define-public geant4-11-2-1
+  (package
+    (inherit geant4-11-2-0)
+    (version "11.2.1")
+    (source
+     (origin
+       (inherit (package-source geant4-11-2-0))
+       (uri (git-reference
+             (url "https://gitlab.cern.ch/geant4/geant4")
+             (commit (string-append "v" version))))
+       (sha256
+        (base32 "1pdg5pf5izd2i05w34jclg0h4hrn3s4mjnr2dvq24lspjisc9r81"))))))
+
+(define-public geant4-vis-11-2-0
+  (package
+    (inherit geant4-vis-11-1-1)
+    (version "11.2.0")
+    (source
+     (origin
+       (inherit (package-source geant4-vis-11-1-1))
```

Comment:
I wonder: why inherit the source at all when it's overridden with a different source anyway?  (Same for all other instances of `package-source` above and below.)